### PR TITLE
Add Text2Video-Zero for video inpainting with ControlNet edge control

### PR DESCRIPTION
**Pull request by Gaohan123**

### CODE REVIEW

**Comment by Gaohan123** [comment]
> As it is a video generation model, please place it in `examples/text2video-zero`

It is a training free method. Basic model and code rely heavily on stable diffusion and controlnet. If we move it to a separate file, there will be a huge amount of repeat code. Besides, I also already prepared an individual tutorial file for it.

**Comment by wtomin** [comment]
Under `text2video_zero/ldm` and `text2video_zero/cldm`, there are many repetitive codes as those in  `stable_diffusion_v2/ldm` and `stable_diffusion_v2/cldm`. Can you use `sys.path.append` to import modules to avoid repeating them?

**Comment by SamitHuang** [comment]
Document writing is to be improved.

File: examples/text2video_zero/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+pip install -r requirements.txt
+```
+
+#### Trained Models
+1. Canny edge maps:
+
+   Please download the trained model [SD1.5-canny-ms checkpoint](https://download.mindspore.cn/toolkits/mindone/stable_diffusion/control_canny_sd_v1.5_static-6350d204.ckpt)
+   It is the pretrained ControlNet[1] with canny edge control, which is one of basic module for Text2Video-Zero.
+
+2. Others:
```
[the remark this answers is not in the document]
Comment (Gaohan123) [reply examples/text2video_zero/README.md]: fixed

File: examples/text2video_zero/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Testing videos preparation
+Please prepare the videos that you want to impaint. There are two examples:
+
+<div align="center">
+<img src="https://github.com/Gaohan123/mindone/assets/20148503/2a460733-3f47-4f04-8a12-c215adfb7536" width="240" height="240" />
+<img src="https://github.com/Gaohan123/mindone/assets/20148503/d21ce9a9-7138-49dd-8275-1a259b2ecf58" width="240" height="240" />
+</div>
+<p align="center">
+  <em> Videos prepared to impaint </em>
```
[the remark this answers is not in the document]
Comment (Gaohan123) [reply examples/text2video_zero/README.md]: fixed

File: examples/text2video_zero/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+   Please download the trained model [SD1.5-canny-ms checkpoint](https://download.mindspore.cn/toolkits/mindone/stable_diffusion/control_canny_sd_v1.5_static-6350d204.ckpt)
+   It is the pretrained ControlNet[1] with canny edge control, which is one of basic module for Text2Video-Zero.
+
+2. Others:
+
+   Coming soon.
+
+Put them under `text2video_zero/models` folder.
+
+#### Testing videos preparation
```
[the remark this answers is not in the document]
Comment (Gaohan123) [reply examples/text2video_zero/README.md]: fixed

File: examples/text2video_zero/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+<div align="center">
+<img src="https://github.com/Gaohan123/mindone/assets/20148503/2a460733-3f47-4f04-8a12-c215adfb7536" width="240" height="240" />
+<img src="https://github.com/Gaohan123/mindone/assets/20148503/d21ce9a9-7138-49dd-8275-1a259b2ecf58" width="240" height="240" />
+</div>
+<p align="center">
+  <em> Videos prepared to impaint </em>
+</p>
+
+Put them in an arbitrary directory on your machine. For example, `text2video_zero/videos`. We already put some in it.
```
[the remark this answers is not in the document]
Comment (Gaohan123) [reply examples/text2video_zero/README.md]: fixed

File: examples/text2video_zero/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Load controls
+1. Canny edge maps:
+   It is implemented with opencv Canny API directly, you don't need to take more actions.
```
[the remark this answers is not in the document]
Comment (Gaohan123) [reply examples/text2video_zero/README.md]: fixed

[the remark this answers is not in the document]
Comment (Gaohan123) [reply examples/text2video_zero/README.md]: fixed

**Review by wtomin — APPROVED** [review]

File: examples/stable_diffusion_v2/controlnet.md [hunk 6]
```diff
@@ -73,49 +73,65 @@ Put them in an arbitrary directory on your machine. For example, `path/to/test_i
...
 
-### Image Generation
+### Parameter Setting
+
+Go to the directory `stable_diffusion_v2/inference` first.
+
 1. Canny edge maps:
 
-   Run the command below to generate images
+   Open the file `stable_diffusion_v2/inference/config/controlnet.yaml`. Set arguments as below:
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/controlnet.md:82]: Please use two separate yaml files for canny control and segmentation control

Comment (Gaohan123) [reply examples/stable_diffusion_v2/controlnet.md]: fixed

File: examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+a_prompt: "best quality, extremely detailed"
+negative_prompt: "longbody, lowres, bad anatomy, bad hands, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality"  # The negative prompt not to render
+H: 512  # Image height, in pixel space
+W: 512  # Image width, in pixel space
+image_resolution: 512
+low_threshold: 100
+high_threshold: 200
+strength: 1
+guess_mode: False
+condition_ckpt_path: "models/deeplabv3plus_s16_ascend_v190_voc2012_research_cv_s16acc79.06_s16multiscale79.96_s16multiscaleflip80.12.ckpt"
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml:13]: can we remote it? canny control doesn't need this ckpt

File: examples/stable_diffusion_v2/inference/config/controlnet_segmentation.yaml
Comment (SamitHuang) [inline examples/stable_diffusion_v2/inference/config/controlnet_segmentation.yaml:1]: deeplabv3 checkpoint should be placed here?

File: examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+image_path: "path/to/test_imgs/dog2.png"
+prompt: "cute dog"  # The prompt to render
+controlnet_mode: "canny" # candidates: [segmentation, canny]
+a_prompt: "best quality, extremely detailed"
+negative_prompt: "longbody, lowres, bad anatomy, bad hands, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality"  # The negative prompt not to render
+H: 512  # Image height, in pixel space
+W: 512  # Image width, in pixel space
+image_resolution: 512
+low_threshold: 100
```
Comment (SamitHuang) [inline examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml:9]: is this thres for segment?

File: examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml (hunk 7 above)
Comment (Gaohan123) [reply examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml]: fixed

File: examples/stable_diffusion_v2/inference/config/controlnet_segmentation.yaml
Comment (Gaohan123) [reply examples/stable_diffusion_v2/inference/config/controlnet_segmentation.yaml:1]: fixed

File: examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml (hunk 8 above)
Comment (Gaohan123) [reply examples/stable_diffusion_v2/inference/config/controlnet_canny.yaml:9]: fixed. removed

**Review by SamitHuang — APPROVED** [review]

**Pull request merged**